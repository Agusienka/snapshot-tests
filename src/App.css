.App {
  text-align: center;
  cursor: pointer;
    user-select: none;
    overflow: hidden;
    background-color: #696a34;
    size: 100%;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: goldenrod;
  font-size: xx-large;
  margin-bottom: 20rem;
  
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.github-card {
  padding-top: 4rem;
  text-align: center;
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: blueviolet;
 background-color: black;
height: 100%;
font-size: 2.1rem;
}


.font-loader {
  font-family: 'Chilanka';
  
}

Button {
color: black;
background-color: red;
size: 3rem;
}
.footer {
  background-color: darkslategrey;
}

.copy {
  size: 100%;
  padding-bottom: 0%;
}