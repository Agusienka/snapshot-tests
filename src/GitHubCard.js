import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import React, { useState } from 'react';
import Alert from 'react-bootstrap/Alert';
import './App.css';



function GitHubCard() {
    const [show, setShow] = useState(true);

  return (
    <>
  
    <div className="copy">
    <div className="github-card" >
    <Card >
      <Card.Img variant="top" src="https://avatars2.githubusercontent.com/u/101678267" />
      <Card.Body>
        <Card.Title className="" style={{fontSize: 45, color: '#b66e5d'}}>AGUSIENKA</Card.Title>
        <Card.Text className="font-loader" style={{color: '#2c9ff7'}}>
        The most unusual item within an armâ€™s reach of me is Dan Moi jaw harp.
        </Card.Text>
        <Alert show={show} variant="success">
        <Alert.Heading style={{color: '#9da462'}}>How's it going?!</Alert.Heading>
        <p style={{color: '#bf9f77'}}>
          Didn't I ask you not to click it? But since you did, have fun with it!
        </p>
        
        <div className="d-flex justify-content-end">
          <Button onClick={() => setShow(false)} variant="outline-success" size="lg">
            Close me y'all!
          </Button>
        </div>
      </Alert>

      {!show && <Button onClick={() => setShow(true)} style={{color: 'white', fontSize: '40px', rounded: true}}>Read Me!!!</Button>}
      
      </Card.Body>
    </Card>
    </div>
    </div>
    <footer className='footer'></footer>
    </>
   
  );
}

export default GitHubCard;