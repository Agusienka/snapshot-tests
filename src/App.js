
import './App.css';
import GitHubCard from './GitHubCard';
import React from 'react';

function App() {
  return (

    <div className="App">
         <h2 style={{color: '#9ba1af', fontSize: '50px'}} className="font-loader">Welcome to my Snapshot-Testing page</h2>
        <a
          className="App-link primary" style={{color: '#9ba1af', fontSize: '30px'}}
          href="https://github.com/Agusienka"
          target="_blank"
          rel="noopener noreferrer"
        >
          GitHub Profile
        </a>
   
    </div>
  );
}


export default App;
