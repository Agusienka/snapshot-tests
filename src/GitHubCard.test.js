import { render, screen, CardImg } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer'
import GitHubCard from './GitHubCard';
import React from 'react';

test('renders a snapshot', () => {
    const tree = renderer.create(<GitHubCard/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  