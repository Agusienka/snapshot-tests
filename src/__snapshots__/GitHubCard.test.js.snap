// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
Array [
  <div
    className="copy"
  >
    <div
      className="github-card"
    >
      <div
        className="card"
      >
        <img
          className="card-img-top"
          src="https://avatars2.githubusercontent.com/u/101678267"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
            style={
              Object {
                "color": "#b66e5d",
                "fontSize": 45,
              }
            }
          >
            AGUSIENKA
          </div>
          <p
            className="font-loader card-text"
            style={
              Object {
                "color": "#2c9ff7",
              }
            }
          >
            The most unusual item within an armâ€™s reach of me is Dan Moi jaw harp.
          </p>
          <div
            className="fade alert alert-success show"
            role="alert"
          >
            <div
              className="alert-heading h4"
              style={
                Object {
                  "color": "#9da462",
                }
              }
            >
              How's it going?!
            </div>
            <p
              style={
                Object {
                  "color": "#bf9f77",
                }
              }
            >
              Didn't I ask you not to click it? But since you did, have fun with it!
            </p>
            <div
              className="d-flex justify-content-end"
            >
              <button
                className="btn btn-outline-success btn-lg"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Close me y'all!
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="footer"
  />,
]
`;
